src/composite.c

```diff
@@ -7,27 +7,28 @@ const int NUM_COLOR_BANDS = 3;
 
 
 /*
-  Composite images `src` and `dst`
+  Composite images `src` and `dst` with premultiplied alpha channel and output
+  image with premultiplied alpha.
  */
-int Composite(VipsObject *context, VipsImage *src, VipsImage *dst, VipsImage **out) {
-  if (src->Bands != 4 || dst->Bands != 4)
+int Composite(VipsObject *context, VipsImage *srcPremultiplied, VipsImage *dstPremultiplied, VipsImage **outPremultiplied) {
+  if (srcPremultiplied->Bands != 4 || dstPremultiplied->Bands != 4)
```

Comment:
FYI, this didn’t require any changes in terms of `vips_premultiply` & Co., since it assumes that the input is premultiplied. I am wondering: Is there a way to store the fact that an image has a premultiplied alpha channel in `VipsImage`?